src/app/domain/backup/services/uploadMedias.ts

```diff
@@ -2,30 +2,77 @@
 import Minilog from '@cozy/minilog'
 
 import { uploadMedia } from '/app/domain/backup/services/uploadMedia'
-import { setMediaAsBackuped } from '/app/domain/backup/services/manageLocalBackupConfig'
-import { Media } from '/app/domain/backup/models/Media'
+import {
+  setBackupAsDone,
+  setMediaAsBackuped
+} from '/app/domain/backup/services/manageLocalBackupConfig'
+import {
+  Media,
+  LocalBackupConfig,
+  ProgressCallback
+} from '/app/domain/backup/models'
+import { getBackupInfo } from '/app/domain/backup/services/manageBackup'
 
 import type CozyClient from 'cozy-client'
+import { IOCozyFile } from 'cozy-client'
 
 const log = Minilog('💿 Backup')
 
+const DOCTYPE_FILES = 'io.cozy.files'
+const DOCTYPE_ALBUMS = 'io.cozy.photos.albums'
+
+let shouldStopBackup = false
+
+export const getShouldStopBackup = (): boolean => {
+  return shouldStopBackup
+}
+
+export const setShouldStopBackup = (value: boolean): void => {
+  shouldStopBackup = value
+}
+
 export const uploadMedias = async (
   client: CozyClient,
-  backupFolderId: string,
-  mediasToUpload: Media[]
+  localBackupConfig: LocalBackupConfig,
+  onProgress: ProgressCallback
 ): Promise<boolean> => {
+  const {
+    remoteBackupConfig: {
+      backupFolder: { id: backupFolderId }
+    },
+    currentBackup: { mediasToBackup: mediasToUpload }
+  } = localBackupConfig
+
   for (const mediaToUpload of mediasToUpload) {
+    if (shouldStopBackup) {
+      shouldStopBackup = false
+      await setBackupAsDone(client)
+      return true
+    }
+
     try {
       const uploadUrl = getUploadUrl(client, backupFolderId, mediaToUpload)
 
-      const { success } = await uploadMedia(client, uploadUrl, mediaToUpload)
+      const { data: documentCreated } = await uploadMedia(
+        client,
+        uploadUrl,
+        mediaToUpload
+      )
+
+      log.debug(`✅ ${mediaToUpload.name} uploaded`)
+
+      await postUpload(
+        client,
+        localBackupConfig,
+        mediaToUpload,
+        documentCreated
+      )
+
+      await setMediaAsBackuped(client, mediaToUpload)
 
-      if (success) {
-        log.debug(`✅ ${mediaToUpload.name} uploaded`)
-        await setMediaAsBackuped(client, mediaToUpload)
-      } else {
-        log.debug(`❌ ${mediaToUpload.name} not uploaded`)
-      }
+      log.debug(`✅ ${mediaToUpload.name} set as backuped`)
+
+      onProgress(await getBackupInfo(client))
```

Comment:
I moved the onProgress callback outside the "upload" try / catch because if it fails here : 
- media has indeed been successfully updated
-  it is strange to fail here so it is better to really throw if something happen here

I also made the debug message more explicit 
`log.debug(
        `❌ ${mediaToUpload.name} not uploaded or set as backuped correctly`
      )`